### PR TITLE
fix: Updated the way the removal logic handles RG-level deployment resolution

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

Updated the way the removal logic handles RG-level deployment resolution
- Before, it would try to resolve deployments, even if (for the current level) it would already know that it won't need to go into any nested deployment
- Also, it could cause the logic to run in an infinite loop as the `deploymentTargets` in a nested run on that level would not be empty, meaning the logic would try to resolve the same object over and over (due to call-by-reference objects)

> **Note:** This change is required for the LZ Accelerator Pattern

## Pipeline Reference

| Pipeline |
| -------- |
|  [avm.res.storage.storage-account](https://github.com/Azure/bicep-registry-modules/actions/workflows/avm.res.storage.storage-account.yml)        |

## Type of Change

- [x] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

**Review by ChrisSidebotham — APPROVED** [review]
Nice 😎

File: avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Get-DeploymentTargetResourceList.ps1 [hunk 1]
```diff
@@ -70,7 +70,7 @@ function Get-DeploymentTargetResourceListInner {
             } else {
                 # In case the resource group itself was already deleted, there is no need to try and fetch deployments from it
                 # In case we already have any such resources in the list, we should remove them
-                [array]$resultSet = $resultSet | Where-Object { $_ -notmatch "\/resourceGroups\/$resourceGroupName\/" }
+                return $resultSet | Where-Object { $_ -notmatch "\/resourceGroups\/$resourceGroupName\/" } | Select-Object -Unique
```
Comment (eriqua) [inline avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Get-DeploymentTargetResourceList.ps1:73]: Probably a good idea to clean up the testing environment before merging

Comment (AlexanderSehr) [inline avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Get-DeploymentTargetResourceList.ps1:73]: Note to self: Run all workflows in fork

Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/helper/Get-DeploymentTargetResourceList.ps1:73]: Triggered all workflows for fork PR https://github.com/Azure/bicep-registry-modules/pull/2198 . Closing

**Pull request closed without merging**